### ISSUE TITLE
Problem for new login user's permission

### ISSUE THREAD

**Issue by oks1998** [snipe/snipe-it#5220]
#### Expected Behavior (or desired behavior if a feature request)

All new user will assign to a group with suppose to view and checkout asset only.
-----

#### Actual Behavior
The new user cannot see any link on the navigation panel, cannot view assets, checkout
-----

#### Please confirm you have done the following before posting your bug report:

- [ V] I have enabled debug mode 
- [V ] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

-----
![image](https://user-images.githubusercontent.com/9070825/37581572-1ba03ae8-2b84-11e8-8f2c-9c0214c9f153.png)
Picture above login by a new user (the user is import from AD but only login first time recently)
In the debug info there, I found snipe didn't run a query that looks like is querying for user group
I ran:
`
composer dump-autoload
composer clear-compiled
php artisan config:clear
php artisan view:clear
php artisan cache:clear
php artisan route:clear
`
but still not working

![image](https://user-images.githubusercontent.com/9070825/37581566-1752e51c-2b84-11e8-93b6-c9f5a5004965.png)
Picture above show a new login user also, but I managed to fix it by assign that user to belong to Superadmin group and change back to normal user group.
And  the user was able to view the asset the link on the side panel.

**Comment by snipe** [comment]
Users are by default not assigned to a group (as AD groups and Snipe-IT groups may or may not match up, and I don't like making assumptions when it comes to permissions.) You can always bulk-edit users to add them to a group.

**Comment by oks1998** [comment]
I use mysql trigger method, to auto insert the user and group id that i want into users_groups table, so it will auto assign a group when new user come in, and the snipeit also show the user is belong to the the group. But the problem is, even i set who ever belong to that group have permission to view asset, and checkout asset, when they login, they cant do anything (just like a user without any permission)

I tried to manually assign a group to a user without the help of trigger of mysql, and the result is the same.

**Comment by snipe** [comment]
What permissions does the group have?

**Comment by oks1998** [comment]
![image](https://user-images.githubusercontent.com/9070825/37635941-bfb348b8-2c39-11e8-8ad4-81fb2abe6a88.png)
only that asset is modified, the rest is remain deny
this is act like a default group when all user newly login into the website, so they can straight away do the job without waiting admin to assign the group

**Comment by oks1998** [comment]
might be, but my issue is new user with pre-assigned group (done using trigger in database) cant "recognise" the belong group (but each user setting got listed out the group that user belongs to) 

only when i assign that user to superadmin, and set it back to group, then that user can only have the correct permission set by the group

**Comment by snipe** [comment]
@oks1998 what's the trigger you're using, exactly? Snipe-IT doesn't have any way of knowing how the group association is being created, it just looks for it when the user logs in.

**Comment by oks1998** [comment]
```
DELIMITER ;;
/*!50003 CREATE*/ /*!50017 DEFINER=`root`@`localhost`*/ /*!50003 TRIGGER AutoAddGroup AFTER INSERT ON users FOR EACH ROW BEGIN
INSERT INTO users_groups (user_id,group_id) VALUES (NEW.id,2);
END */;;
DELIMITER ;
```

so it detect when new user record is added to table "users", then the trigget will fired and a new record with new user id and the group id that i want will added to "users_groups"

**Comment by oks1998** [comment]
any idea what is the cause of issue?

**Comment by snipe** [comment]
There's nothing on the code end that can tell whether that record was created by your trigger or by our system though, so I don't have any idea why you're seeing what you're seeing.

**Comment by vogtmh** [comment]
Hi oks1998,

can you tell me more about how you created the trigger? I tried yours and it just doesn't work. The new.id seems to be inaccessible as it's an auto-increment.

**Comment by oks1998** [comment]
sorry for late reply, but does your "users_groups" 's user_id is auto increment? that cant be right? that table is for assigning user to belong to chosen group right?

**Comment by SpearRIT** [comment]
I am seeing this exact issue on Version v4.8.0 - build 4186 (master) (upgraded through several versions since mid-2018) with the same exact workaround. However, I am not doing a SQL trigger, simply importing users directly from LDAP.

As seen in the images on this ticket, when the user first logs in there is no sidebar and they receive "Sad Panda 403" errors when trying to get to pages like /hardware. The users are ONLY able to view their own user page.

The fact the adding Super User permissions and then taking them away without making any other permissions changes to the user or the group lead me to believe that this is a permissions issue somewhere.

**Comment by abby0725** [comment]
![Capture](https://user-images.githubusercontent.com/61541532/76054774-bbeb6000-5fac-11ea-9e13-63e212813f23.PNG)

Hi @SpearRIT , u mean deny the "Super User" permission?

**Comment by SpearRIT** [comment]
Hi @abby0725, 

More explicitly, I "Grant" Super User permissions on the individual user (under People), save the change, edit the user again, and set the Super User permission to "inherit".

Hope that helps!

**Comment by abby0725** [comment]
Hi @SpearRIT ,
I still not understand. can you do a simple screenshot to guide me? As I follow it, it still unable to view ALL the asset created.
![1](https://user-images.githubusercontent.com/61541532/76206204-8c4f8880-6236-11ea-8622-ed73c1ec256a.PNG)
![2](https://user-images.githubusercontent.com/61541532/76206209-8d80b580-6236-11ea-9fb5-8c6a0b8bb2bc.PNG)

**Comment by SpearRIT** [comment]
@abby0725 First, I define the group permissions: example below.
![image](https://user-images.githubusercontent.com/38700216/76223492-cd717980-61f1-11ea-879b-c023cac19ef2.png)
Then I got into the individual user and grant Global Super User:
![image](https://user-images.githubusercontent.com/38700216/76224433-145f6f00-61f2-11ea-92dc-4bcd304c7452.png)
I save the user, then set it back to inherit:
![image](https://user-images.githubusercontent.com/38700216/76224720-2a6d2f80-61f2-11ea-8e43-de9114ab4c14.png)
Then save the user again.
I also confirm that the user is part of the appropriate group.

That's the complete fix that has worked for me. Once that is done, users are able to view all assets by clicking "Assets" -> "List all". Please note that the group permissions must be set correctly for this to work.

**Comment by abby0725** [comment]
Hi, i still unable to set the user who can view ALL asset. This is what I set up.
1. Define the group permission
![p1](https://user-images.githubusercontent.com/61541532/76399096-0445bc00-63b9-11ea-90b9-ea17852d22cf.PNG)
2. Create a user with the group permission correctly.
![p2](https://user-images.githubusercontent.com/61541532/76399144-19bae600-63b9-11ea-87ae-4aed71527b7c.PNG)
3. As I follow your step, go into the individual user and grant global super user. then, save.
![p3](https://user-images.githubusercontent.com/61541532/76399207-35be8780-63b9-11ea-9117-768c880be1a7.PNG)
4. set it back to inherit. Then save.
![p4](https://user-images.githubusercontent.com/61541532/76399391-7a4a2300-63b9-11ea-9e21-90b371a87f55.PNG)
5. When I login, it only can view the asset which assigned for the user.
![p5](https://user-images.githubusercontent.com/61541532/76399444-8e8e2000-63b9-11ea-97e8-6a2841bce0c5.PNG)
6. When I click list ALL, nothing show
![p6](https://user-images.githubusercontent.com/61541532/76399525-b4b3c000-63b9-11ea-8a9f-dc726997b50e.PNG)
7. Show ALL asset when i login with global super user account
![p7](https://user-images.githubusercontent.com/61541532/76399901-6652f100-63ba-11ea-936d-bdd6bdda30f9.PNG)

**Comment by abby0725** [comment]
Is there got any way to only view ALL asset without the permission of edit and delete?

**Comment by abby0725** [comment]
@SpearRIT

**Comment by SpearRIT** [comment]
@abby0725 Are you assets flagged as requestable?

![image](https://user-images.githubusercontent.com/38700216/76479783-4b48a700-63e3-11ea-88c1-17585e8af67b.png)

Also, if you create an asset for that user, do they see it in both views?

I'm not an expert on the software, just reporting what worked for me.

**Comment by abby0725** [comment]
@SpearRIT ya. I set all asset as requestable. 
![prob1](https://user-images.githubusercontent.com/61541532/76480105-c09a9300-6448-11ea-988f-d45ed95e23d4.PNG)

Still unable to view ALL asset when i didn't have super user permission.
![Prob2](https://user-images.githubusercontent.com/61541532/76480227-153e0e00-6449-11ea-882d-0fc44da96d30.PNG)

**Comment by SpearRIT** [comment]
@abby0725 Does the behavior change if you grant all "asset" permissions including edit and delete?

Did anyone make changes per this page? https://snipe-it.readme.io/docs/user-permissions

<blockquote><img src="/favicon.ico" width="48" align="right"><div>Snipe-IT Documentation</div><div><strong><a href="https://snipe-it.readme.io/docs/user-permissions">User Permissions</a></strong></div><div>Snipe-IT uses Policies to determine whether a user should be able to view, edit, delete, etc a resource.  The policies are location in app/Policies, and extend from app/Policies/SnipePermissionsPolicy.php, which does the more complex work of determining whether or not a user should be able to perfor...</div></blockquote>

**Comment by abby0725** [comment]
@SpearRIT when I grant all "asset" permission, it also only can view the asset which assigned for the user. NOT ALL asset. If I am the global super user, I just can view ALL asset. Is it the system normal behaviour? or anything wrong with it? 
![issue2](https://user-images.githubusercontent.com/61541532/76581458-55b38f80-650e-11ea-9754-3a95999ce793.PNG)
![issue1](https://user-images.githubusercontent.com/61541532/76581460-577d5300-650e-11ea-897a-7b4a4fbdf4ec.PNG)

**Comment by ipaqmaster** [comment]
I've got a recently-upgraded Snipe-IT installation (Some 2018 commit>f004007c) which has been configured for LDAP with an `LDAP Filter` which only matches user accounts which are members of a specific SnipeIT access AD group. All our IT team members reside within that group.

Not that the name is related related in any way - but I made a group in Snipe-IT of the same name with `Superuser` checked. I set this group as the `LDAP Permissions Group`.

Logging in for the first time using my LDAP credentials opens up to `/account/view-assets`. Evidently my account has been given no permissions whatsoever indicating the `LDAP Permissions Group` feature has failed. 

I'm not sure how to proceed. If our IT staff can't log into Snipe-IT with a predetermined level of access via the `LDAP Permissions Group` configuration line.  I don't understand what LDAP is going to prove useful for here.
